lib/bunch/sref.ex

```diff
@@ -0,0 +1,32 @@
+defmodule Bunch.Sref do
+  @moduledoc """
+  A bunch of stuff for shortly-inspectable references.
+  """
+  @enforce_keys [:ref, :hash]
+  defstruct @enforce_keys
+
+  @type t :: %__MODULE__{ref: reference, hash: String.t()}
+
+  @doc """
+  Creates a shortly-inspectable reference.
+
+      iex> IEx.Helpers.ref(0, 1, 2, 3) |> #{inspect(__MODULE__)}.new() |> inspect()
+      "#6f1ef15f"
+      iex> <<"#", hash::binary-size(8)>> = #{inspect(__MODULE__)}.new() |> inspect()
+      iex> Base.decode16(hash, case: :lower) |> elem(0)
+      :ok
+
+  """
+  @spec new(reference) :: t
+  def new(ref \\ make_ref()) do
+    '#Ref' ++ ref_list = :erlang.ref_to_list(ref)
+    <<bin_hash_part::binary-size(4), _::binary>> = :crypto.hash(:sha, ref_list)
+    hash = "#" <> Base.encode16(bin_hash_part, case: :lower, padding: false)
```

Comment:
encode16 doesn't have `padding` option

Reply:
BTW, why encode with base 16? using base64 you would get more condensed representation

Reply:
I relied on the way Git displays commits' references. Also, with base64 we'd gain only two characters, while it's alphabet is harder to reason about than just hex imho